### PR TITLE
Feature validation

### PR DESCRIPTION
**Pull request by Tchilo**
# Add the following validations:
 ## For the User model:
   - Name must not be blank.
   - PostsCounter must be an integer greater than or equal to zero.
  ## For the Post model:
   - Title must not be blank.
   - Title must not exceed 250 characters.
  - CommentsCounter must be an integer greater than or equal to zero.
  - LikesCounter must be an integer greater than or equal to zero.
   - Add unit specs for all models' methods and validations.
  ### Add flash messages in the create actions in all controllers.
  ###  Use the N+1 problem when fetching all posts and their comments for a user.

### CODE REVIEW

File: spec/models/comment_spec.rb [hunk 1]
```diff
@@ -1,5 +1,39 @@
...
+#   before { subject.save }
+
+#   it 'should return most recent comments' do
+#     user = User.new(name: 'Lenny', photo: 'image', bio: 'bio', posts_counter: 0)
+#     subject.comments.create!(user: user, text: 'Well said.')
+#     subject.comments.create!(user: user, text: 'I would not have said it better')
+
+#     expect(subject.comments_counter).to eql 2
+#   end
+# end
```
Comment (elisha2kyakpo1) [inline spec/models/comment_spec.rb]: - [x] Kindly remove all the commented code to make the code more clean and readable.

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 @Tchilo,

You have done a great job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

### To highlight

- The PR title and description look good 👍
- No linter errors 👍
- Good commit messages 👍

## Required Changes ♻️
_Check the comments under the review._

- [x] Kindly fix the N + 1 query problem in a post controller, there are so many options available. For example, we can do this `@posts = @user.posts.includes(:comments)` in the index action to fetch user posts together with their comments.
- [x] Let's include tests for model validations as well, something like this below:
```
describe 'validations' do
  it { should validate_presence_of(:name) }
end
```


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/models/comment_spec.rb (hunk 1 above)
Comment (Tchilo) [reply spec/models/comment_spec.rb]: Many thanks for the code review I have made the needed changes

**Review by elisha2kyakpo1 — APPROVED** [review]
Hi 👋 @Tchilo,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

You have done a great job making the changes requested ✔️

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**